### DISCUSSION TITLE
Docker-compose file yaml syntax error

### DISCUSSION THREAD

**Discussion by dinotheextinct** [felddy/foundryvtt-docker#381] [Q&A] [answered]
# 🐛 Bug Report #
docker-compose.yml not working, problem with service mapping, following error message 
`services.foundry.environment must be a mapping`

``` 
  1 ---
  2 version: "3.8"
  3
  4 secrets:
  5   config_json:
  6     file: geheimnisse.json
  7
  8 services:
  9   foundry:
 10     image: felddy/foundryvtt:release
 11     hostname: saurierfriedhof
 12     init: true
 13     volumes:
 14       - type: bind
 15         source: ./foundry_data
 16         target: /data
 17     environment:
 18     ports:
 19       - target: 30000
 20         published: 30000
 21         protocol: tcp
 22     secrets:
 23       - source: config_json
 24         target: config.json
```

## To Reproduce ##

write docker-compose.yml and try to start it with docker compose up --detach

## Expected behavior ##
Installing starting the docker container through the compose file

**Comment by felddy** [comment, marked as answer]
They `environment` mapping (line 17) is empty.  Deleting it will correct the error you are receiving from `docker`. 

See: 
- https://ota-meshi.github.io/eslint-plugin-yml/rules/no-empty-mapping-value.html

**Comment by dinotheextinct** [comment]
Alright thanks, I know got at least the compose file firing off, but no docker seems to be started with docker ps. Although I get `[+] Running 1/1
 ⠿ Container foundry-docker-foundry-1  Started   `
